spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java

```diff
@@ -0,0 +1,117 @@
+
+	// not using currently but expect to include global properties here if
+	// required
+	private final KinesisBinderConfigurationProperties configurationProperties;
+
+	// not currently utilising any extended properties - not sure if I will need
+	// these
+	private KinesisExtendedBindingProperties extendedBindingProperties = new KinesisExtendedBindingProperties();
+
+	private AmazonKinesisAsync amazonKinesis;
+
+	public KinesisMessageChannelBinder(KinesisBinderConfigurationProperties configurationProperties,
+			KinesisTopicProvisioner provisioningProvider) {
+		super(false, null, provisioningProvider);
+		this.configurationProperties = configurationProperties;
+
+		this.amazonKinesis = AmazonKinesisAsyncClientBuilder.defaultClient();
+
+	}
+
+	@Override
+	public KinesisConsumerProperties getExtendedConsumerProperties(String channelName) {
+		return this.extendedBindingProperties.getExtendedConsumerProperties(channelName);
+	}
+
+	@Override
+	public KinesisProducerProperties getExtendedProducerProperties(String channelName) {
+		return this.extendedBindingProperties.getExtendedProducerProperties(channelName);
+	}
+
+	// below are the main methods to implement - these will create the message
+	// handlers used by the application
+	// to put and consume messages
+	@Override
+	protected MessageHandler createProducerMessageHandler(ProducerDestination destination,
+			ExtendedProducerProperties<KinesisProducerProperties> producerProperties) throws Exception {
+
+		KinesisMessageHandler kinesisMessageHandler = new KinesisMessageHandler(amazonKinesis);
+
+		kinesisMessageHandler.setStream(destination.getName());
+		// tidy up partition key
+		kinesisMessageHandler.setPartitionKey("name");
+		kinesisMessageHandler.setBeanFactory(this.getBeanFactory());
+
+		return kinesisMessageHandler;
+	}
+
+	@Override
+	protected MessageProducer createConsumerEndpoint(ConsumerDestination destination, String group,
+			ExtendedConsumerProperties<KinesisConsumerProperties> properties) throws Exception {
+
+		KinesisMessageDrivenChannelAdapter adapter = new KinesisMessageDrivenChannelAdapter(amazonKinesis,
+				destination.getName());
+
+		adapter.setOutputChannel(kinesisChannel());
+		adapter.setCheckpointStore(checkpointStore());
+
+		// explicitly setting the offset - LATEST is the default but added here
+		// so can be configured later
+		adapter.setStreamInitialSequence(KinesisShardOffset.latest());
+		// need to move these properties to the appropriate properties class
+		adapter.setCheckpointMode(CheckpointMode.record);
+		adapter.setListenerMode(ListenerMode.record);
+		adapter.setStartTimeout(10000);
+		adapter.setDescribeStreamRetries(1);
+		adapter.setConcurrency(10);
+
+		return adapter;
+	}
+
+	public PollableChannel kinesisChannel() {
+		return new QueueChannel();
+	}
+
+	public MetadataStore checkpointStore() {
+		SimpleMetadataStore simpleMetadataStore = new SimpleMetadataStore();
```

Comment:
This definitely has to be replaced to the `DymanoDbMetadaStore`, when it is available: https://github.com/spring-projects/spring-integration-aws/issues/64